oshi-core/src/main/java/oshi/hardware/common/AbstractPowerSource.java

```diff
@@ -46,11 +55,27 @@
      * @param newTimeRemaining
      *            Seconds of time remaining
      */
+    @Deprecated
     public AbstractPowerSource(String newName, double newRemainingCapacity, double newTimeRemaining) {
         this.name = newName;
         this.remainingCapacity = newRemainingCapacity;
         this.timeRemaining = newTimeRemaining;
     }
+    
+    /**
+     * Super constructor used by platform specific implementations of
+     * PowerSource. Initializes all values to defaults, which are then set
+     * using the setter methods.
+     */
+    public AbstractPowerSource() {
+        this.name = "unknown";
+        this.remainingCapacity = 0;
+        this.timeRemaining = -1;
+        this.health = 1d;

```

Comment:
OK, I'm really being type A and this is probably so trivial I shouldn't comment.  But I'm on a roll. ;)   This declaration is properly a `double`.  The other defaults above and below should also match their defined type to save a nanosecond or two of int-to-double casting.  Or not.  Not a big deal, really.  ;)